CRM/Signatures/Upgrader.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/*------------------------------------------------------------+
+| SYSTOPIA Signatures                                         |
+| Copyright (C) 2017 SYSTOPIA                                 |
+| Author: B. Endres (endres@systopia.de)                      |
+|         J. Schuppe (schuppe@systopia.de)                    |
++-------------------------------------------------------------+
+| This program is released as free software under the         |
+| Affero GPL license. You can redistribute it and/or          |
+| modify it under the terms of this license which you         |
+| can read by viewing the included agpl.txt or online         |
+| at www.gnu.org/licenses/agpl.html. Removal of this          |
+| copyright header is strictly prohibited without             |
+| written permission from the original author(s).             |
++-------------------------------------------------------------*/
+
+use CRM_Signatures_ExtensionUtil as E;
+
+/**
+ * Collection of upgrade steps.
+ */
+class CRM_Signatures_Upgrader extends CRM_Signatures_Upgrader_Base {
+
+  /**
+   * Split the signatures settings into contact-specific settings.
+   *
+   * @return bool
+   *   Whether the upgrade task executed successfully.
+   */
+  public function upgrade_4200() {
```

Comment:
Why starting so high?

Reply:
The docs say this naming schema follows what Drupal's `hook_update_N()` does: first digit for core version compatibility, second for extension major version, 3rd and 4th digits for internal numbering. Turns out, there is just a `current_revision < max(all_revisions)` comparison. Changed that to `4000`.